ui/src/app/base/sagas/actions.ts

```diff
@@ -0,0 +1,127 @@
+import type { PayloadAction } from "@reduxjs/toolkit";
+import { call } from "redux-saga/effects";
+import type { SagaGenerator } from "typed-redux-saga";
+
+import type { WebSocketClient } from "../../../websocket-client";
+
+import type { TSFixMe } from "app/base/types";
+import { actions as domainActions } from "app/store/domain";
+import type { UpdateRecordParams } from "app/store/domain/types";
+import { isAddressRecord } from "app/store/domain/utils";
+import { actions as machineActions } from "app/store/machine";
+import type { Machine } from "app/store/machine/types";
+import { actions as resourcePoolActions } from "app/store/resourcepool";
+import type {
+  CreateWithMachinesParams,
+  ResourcePool,
+} from "app/store/resourcepool/types";
+
+export type MessageHandler = {
+  action: string;
+  method: (...args: unknown[]) => unknown;
+};
+
+/**
+ * Generate functions that will use the response to create the dispatchable
+ * action to set the pool for each machine.
+ * @param machineIDs - A list of machine ids.
+ * @returns The list of action creator functions.
+ */
+export const generateMachinePoolActionCreators: TSFixMe = (
+  machineIDs: Machine["system_id"][]
+) =>
+  machineIDs.map(
+    (machineID) => (result: ResourcePool) =>
+      machineActions.setPool({ systemId: machineID, poolId: result.id })
+  );
+
+/**
+ * Handle creating a pool and then attaching machines to that pool.
+ * @param socketClient - The websocket client instance.
+ * @param sendMessage - The saga that handles sending websocket messages.
+ * @param action - The redux action with pool and machine data.
+ */
+export function* createPoolWithMachines(
+  socketClient: WebSocketClient,
+  sendMessage: TSFixMe,
+  { payload }: PayloadAction<{ params: CreateWithMachinesParams }>
+): SagaGenerator<void> {
+  const { machineIDs, pool } = payload.params;
+  const actionCreators = yield call(
```

Comment:
To have these return the correct types we need to use `yield* call` from `import { call } from "typed-redux-saga/macro";`